extensions/migrate.md

```diff
@@ -0,0 +1,321 @@
+# IRCv3 `migrate` Extension
+
+*Copyright &copy; 2016 Alex Iadicicco <https://github.com/aji>*
+
+*Copyright &copy; 2016 Justin Kaufman <https://github.com/akaritakai>*
+
+*Unlimited redistribution and modification of this document is allowed provided
+that the above copyright notice and this permission notice remains intact*
+
+This document describes the `migrate` IRCv3 client capability and associated
+protocol, which allows servers to migrate client sessions between servers, with
+appropriate client cooperation.
+
+Clients that support the migration protocol outlined in this document should
+register the `migrate` client capability. If the server supports client
+migrations (even if the network is not in a configuration to support
+migrations), it should accept the capability.
+
+# Protocol overview
+
+From the client's perspective, a migration works as follows. The details of this
+protocol are described later in this section.
+
+1. The server sends `MIGRATE START` to indicate that it wants to migrate the
+   client to a different server. Included in this message are a *resume token*
+   and *confirm token* that will be used when connecting to the new server.
```

Comment:
You should specify the format of the message with these two tokens. I think it is rather clear what you mean, but since you are writing a specification, it is better to be explicit.